route/car-route.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const Router = require('express').Router;
+const jsonParser = require('body-parser').json();
+const Car = require('../model/car.js');
+const debug = require('debug')('car:car-route ');
+const carRouter = module.exports = new Router();
+
+carRouter.post('/api/car', jsonParser, function(req, res, next){
+  debug('POST: /api/car');
+  req.body.timestamp = new Date();
+  new Car(req.body).save()
+  .then( car => res.json(car))
+  .catch( err => next(err));
+});
+
+carRouter.get('/api/car/:id', function(req, res, next) {
+  debug('GET: /api/car/:id');
+  Car.findById(req.params.id)
+  .then( car => res.json(car))
+  .catch( err => next(err));
+});
+
+carRouter.get('/api/car', function(req, res, next){
+  debug('GET: /api/car');
+  Car.findById()
+  .then( car => res.json(car))
+  .catch( err => next(err));
+});
+
+carRouter.put('/api/car/:id', jsonParser, function(req, res, next){
+  debug('PUT: /api/car/:id');
+  Car.findByIdAndUpdate(req.params.id, req.body, {new: true})
+  .then( car => res.json(car))
+  .catch( err => next(err));
+});
+
+carRouter.put('/api/car', jsonParser, function(req, res, next){
+  debug('PUT: /api/car');
+  Car.findByIdAndUpdate(req.params.id, req.body, {new: true})
+  .then( car => res.json(car))
+  .catch( err => next(err));
+});
```

Comment:
Did you push your changes?  I don't see it here. I'd have to see what the error looks like in my terminal.  I think most of the errors I saw from you were the 'expected 500 to equal 404' or similar type errors.  That issue is the one I was referring to with the asterisks comment on line 35.  It usually says something like 'validation error'.